addons/mail/static/src/attachments/attachment_viewer.js

```diff
@@ -67,40 +67,47 @@ export class AttachmentViewer extends Component {
     }
 
     onKeydown(ev) {
-        switch (ev.key) {
-            case "ArrowRight":
-                this.next();
-                break;
-            case "ArrowLeft":
-                this.previous();
-                break;
-            case "Escape":
-                this.close();
-                break;
-            case "q":
-                this.close();
-                break;
-            case "r":
-                this.rotate();
-                break;
-            case "+":
-                this.zoomIn();
-                break;
-            case "-":
-                this.zoomOut();
-                break;
-            case "0":
-                this.resetZoom();
-                break;
-            default:
-                return;
+        if (this.handleEvent("navigation", ev)) {
```

Comment:
This function always returns `true` for `"navigation"`. This conditional is useless, so would just drop it

Reply:
It's best to intercept keydown in `input` and `stopPropagation()`, so it's not intercepted by `AttachmentViewer@onKeydown`. This also allow to not define this `handleEvent` function that looks strange by itself.

Reply:
I didn't see that there was no issue for the wheel and mouseDown events. So we indeed don't need the `handleEvent` function. But the input stopPropagation doesn't seem to work.
But I can simply do this in documents 
```js
    onKeydown(ev) {
        if (ev.target.closest(".o_documents_view")) {
            super.onKeydown(ev);
        }
    }
```

What do you think ?

Reply:
The `AttachmentViewer@onKeydown` is handled on `document` in bubbling phase, so I would expect `ev.stopPropagation()` in bubbling phase handling of `keydown` in the input to work (it's more specific, and the `stopPropagation()` should prevent being it reached by `document.keydown` from the `AttachmentViewer@onKeydown`.

I'm good with patching `onKeydown` as you did as a quick fix. I'd at least put a brief comment on why `onKeydown` is ignored when outside of the viewer. Thanks!

Reply:
Ok, then patching `onKeydown` in documents is best tradeoff!

Reply:
Wait... `useExternalListener('keydown')` is on `document`. This assumes the `AttachmentViewer` is fullscreen, which is the case in all but documents.

We could have checked `ev.target.closest('.o-mail-AttachmentViewer')` in `mail/` and this would not need any patch in document at all.

Reply:
Actually when clicking on an attachment in a chatter 'outside documents), the keydown event is triggered on the `.o-mail-AttachmentImage` which is outside the `.o-mail-AttachmentViewer`. So you would need to click on the full screen image first.

Reply:
Oh I see... probably the focus could move to the attachment viewer when it opens.
Both solutions require few LOCs anyway, and the current solution with patching in document is small enough, so better keep that solution and merge asap. Thanks!